idaes/generic_models/properties/core/eos/ideal.py

```diff
@@ -299,6 +300,20 @@ def gibbs_mol_phase_comp(b, p, j):
                 b.entr_mol_phase_comp[p, j] *
                 b.temperature)
 
+    @staticmethod
+    def pressure_osmotic_phase(b, p):
+        try:
+            solvent_set = b.params.solvent_set
+        except AttributeError:
+            raise ConfigurationError(
+                f"{b.name} called for pressure_osmotic, but no solvents were "
+                f"defined. Osmotic pressure requires at least one component "
+                f"to be declared as a solvent.")
+        C = sum(b.conc_mol_phase_comp[p, j] for j in b.component_list
```

Comment:
The issue here is how the user represents ions. This is the sum across all non-solvent species, so if the user declared the ions in some form then they will be implicitly included. However, if you just declare NaCl and expect that to work... I am not even sure how we can catch that condition without adding very case specific arguments.

Reply:
Should we think about making changes to the NaCl property model in Proteuslib based on what you have here, or do you both think the NaCl model can remain independent of this added functionality?

Reply:
@andrewlee94 what if we add an attribute to the solute component that notes the number of dissociated ions?

Reply:
@adam-a-a the NaCl property model won't use these methods, its an independent property package.

Reply:
@TimBartholomew That is only applicable if the Solute is a salt however. Remember that the Ion components derive from Solute, and number of dissociated ions makes no sense there. Seeing as we probably want to support this use case, I see two options:

1. Make `Apparent` work for non-electrolyte systems and make users define salts using this component (which makes sense),
2. Define a new `Salt` component which handles this, but which only works for non-electrolyte systems.

Writing that out, 1 would seem to be the only sane answer.

@adam-a-a The NaCL and seawater property packages are custom-coded so they can do what they want. However, it would make sense to make this a new standard for aqueous systems and have them define the `solute_set` and `solvent_set` as well.

Reply:
I've updated `Apparent` so that it should work with non-electrolyte systems and include the `i` term when calculating osmotic pressure.